36-vending-machine-monitor/firmware/vending-machine/notecard-sleep.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "app.h"
+
+/**
+ * @brief Puts notecard and the host to sleep for a given period.
+ * This is used to conserve power when running on battery power.
+ * Once enabled, the host will wait for a delay period before sleeping, to avoid unnecessary
+ * shutdowns on short power outages. If power doesn't resume, then 
+ */
+class NotecardSleep {
+
+    uint32_t delayBeforeSleep;
+    uint32_t sleepPeriod;
+    bool enabled;
+    uint32_t enabledAt;
+
+    void sendSleepRequest(uint32_t duration) {
+        J* req = notecard.newCommand("card.attn");
+        JAddStringToObject(req, "mode", "sleep,usb");
+        JAddNumberToObject(req, "seconds", duration/1000);
```

Comment:
Can duration just be in seconds? I don't see the value in having it in milliseconds if the Notecard only accepts the seconds argument as an int32_t.